classes/local/api/tracker.php

```diff
@@ -37,13 +37,55 @@
  * @copyright  Talview, 2020
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-class tracker {
+class tracker
+{
     /**
      * Insert the actual tracking code.
      *
      * @return void As the insertion is done through the {js} template API.
      */
-    public static function insert_tracking() {
+
+    private static function generate_auth_token($api_base_url, $payload)
+    {
+        $curl = curl_init();
+        curl_setopt_array($curl, [
+            CURLOPT_URL => $api_base_url . '/auth',
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_ENCODING => "",
+            CURLOPT_MAXREDIRS => 10,
+            CURLOPT_TIMEOUT => 30,
+            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+            CURLOPT_CUSTOMREQUEST => "POST",
+            CURLOPT_POSTFIELDS => json_encode($payload),
+            CURLOPT_HTTPHEADER => [
+                "Content-Type: application/json"
+            ],
+        ]);
+        try {
+            $response = curl_exec($curl);
+            $err = curl_error($curl);
+            $httpcode = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+            curl_close($curl);
+            if ($err) {
+                throw new CustomException($err);
+            } elseif ($response && $httpcode != 200) {
+                throw new CustomException($response);
+            } else {
+                return $response;
+            }
+        } catch (\Throwable $err) {
+            self::capture_error($err);
+        }
+    }
+
+
+   private static function capture_error(\Throwable $err)
+    {
+        \Sentry\init(['dsn' => 'https://61facdc5414c4c73ab2b17fe902bf9ba@o286634.ingest.sentry.io/5304587']);
```

Comment:
@rohansharmasitoula Create backlog tickets to move DSN to constant file and then to GH Secrets, when moving to gh secrets ensure to deprecate this DSN and get a new DSN created